src/main/java/blusunrize/immersiveengineering/api/energy/DieselHandler.java

```diff
@@ -24,66 +24,52 @@
  */
 public class DieselHandler
 {
-	static final HashMap<ResourceLocation, Integer> dieselGenBurnTime = new HashMap<>();
-	static final Set<Fluid> drillFuel = new HashSet<>();
+	static final HashMap<Tag<Fluid>, Integer> dieselGenBurnTime = new HashMap<>();

```

Comment:
Can we change the type of the field to `Map<…>` instead of `HashMap<…>`? I'm not sure whether we actually still want a `Map` here in any case, since we only ever iterate over it. Maybe `List<Pair<…>>` would be more appropriate?

Reply:
Yeah we are using it like a list, you make a good point.